addons/mail/models/mail_channel.py

```diff
@@ -820,14 +820,19 @@ def get_mention_suggestions(self, search, limit=8):
     @api.model
     def channel_fetch_listeners(self, uuid):
         """ Return the id, name and email of partners listening to the given channel """
-        self._cr.execute("""
+        where_clause, where_params = self._channel_fetch_listeners_where_clause(uuid)
```

Comment:
After discussing with Antony, he prefers that people override the whole method instead of having too much sub-methods. So here we should keep the method as it is and override it completely in mail_bot to write the new SQL that applies without calling super. This is less modular in this case we prefer having readability on modularity.